### PR TITLE
PWN-3105 - koin graph tests

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3105

## Description of Work 
made a test to check koin graph made properly

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/App.kt [hunk 1]
```diff
@@ -105,4 +115,16 @@ class App : Application() {
...
         Timber.plant(TimberCrashTree(crashLoggingService))
     }
+
+    private fun setupCrashLoggingService() {
+        crashLoggingService.apply {
+            isLoggingEnabled = BuildConfig.CRASHLYTICS_ENABLED
+            setCustomKey(BuildConfig.TASK_NUMBER, "")
+
+            val currentUserPublicKey = kotlin.runCatching { tokenKeyProvider.publicKey }
+            setUserId(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/App.kt:125]: i think I missed to register a listener here, do mind fixing it?

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/notification/AppNotificationManager.kt [hunk 2]
```diff
@@ -35,7 +36,7 @@ class AppNotificationManager(
         )
     }
 
-    private val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+    private val notificationManager = context.getSystemService<NotificationManager>()!!
```
Comment (davran1001) [inline app/src/main/java/org/p2p/wallet/notification/AppNotificationManager.kt:39]: Are you sure that it cannot be nullable ?

File: app/src/main/java/org/p2p/wallet/updates/ConnectionStateProvider.kt [hunk 3]
```diff
@@ -12,7 +13,7 @@ import kotlinx.coroutines.flow.onCompletion
 import kotlinx.coroutines.flow.onStart
 
 class ConnectionStateProvider(context: Context) {
-    private val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    private val connectivityManager: ConnectivityManager = context.getSystemService()!!
```
Comment (davran1001) [inline app/src/main/java/org/p2p/wallet/updates/ConnectionStateProvider.kt:16]: i think we should to avoid using !!

**Review by davran1001 — APPROVED** [review]

**Review by davran1001 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/notification/AppNotificationManager.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/notification/AppNotificationManager.kt:39]: context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager 
== 
context.getSystemService<NotificationManager>()!!

these lines throw NPE nevertheless

File: app/src/main/java/org/p2p/wallet/updates/ConnectionStateProvider.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/updates/ConnectionStateProvider.kt:16]: agreed, do you have an alternative how to get System Services?

**Review by kuznetsov — APPROVED** [review]

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/App.kt [hunk 4]
```diff
@@ -105,4 +115,18 @@ class App : Application() {
...
+    private fun setupCrashLoggingService() {
+        crashLoggingService.apply {
+            isLoggingEnabled = BuildConfig.CRASHLYTICS_ENABLED
+            setCustomKey(BuildConfig.TASK_NUMBER, "")
+
+            val currentUserPublicKey = kotlin.runCatching { tokenKeyProvider.publicKey }
+            setUserId(CrashLoggingService.UserId(currentUserPublicKey.getOrDefault("")))
+
+            tokenKeyProvider.registerListener { newUserPublicKey ->
+                setUserId(CrashLoggingService.UserId(newUserPublicKey))
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/App.kt]: mockked

**Review by 1sakov — APPROVED** [review]

**Pull request merged**